integration/integration_test.go

```diff
@@ -109,6 +115,45 @@ func cleanupFunc(cleanupCommands []*command) {
 	}
 }
 
+func expectAtLeastOneMatch[T any](output string, expected T, match func(T) bool) error {
+	scanner := bufio.NewScanner(strings.NewReader(output))
+
+	for scanner.Scan() {
+		var obj T
+
+		line := scanner.Text()
+		if err := json.Unmarshal([]byte(line), &obj); err != nil {
+			return fmt.Errorf("unmarshaling line %q: %w", line, err)
+		}
+
+		// Can't use reflect.DeepEqual as we don't have all the expected fields.
+		// For now, each test verifies manually their subset of expected fields.
```

Comment:
This is the main problem with this approach. My first idea is to use this custom match where each test verifies its known fields. WDYT?

Reply:
What about removing the fields we can't compare from the event and using reflect.DeepEqual -> https://github.com/kinvolk/inspektor-gadget/commit/3dbf544a05f05e972f5a284d5c1a1fcb976decc9. The resulting code is shorter and we' could provide detailed information of which is the field not matching by using `cmp.Diff()`.

Reply:
Great suggestion, looks better!